### ISSUE TITLE
It can't build ios-framework when I use plugin.

### ISSUE THREAD

**Issue by mynameissujie** [flutter/flutter#48859]
## command

flutter build ios-framework --output=build_ios_test

## exception

FileSystemException: FileSystemException: Directory listing failed, path = 'build_ios_test/Debug/iphoneos/Debug-iphoneos/' (OS Error: No such file or directory, errno = 2)

**Comment by mynameissujie** [comment]
[32m[✓][39m Flutter (Channel beta, v1.12.13+hotfix.6, on Mac OS X 10.14.4 18E226, locale
    zh-Hans-CN)

**Comment by Vanethos** [comment]
Could you provide the full error logs?

And the full output of `flutter doctor`?

**Comment by mynameissujie** [comment]
when I use my plugin uploaded to git,it can run well in runner.workspace .But when flutter build ios-framework.it goes wrong.
Unable to build plugin frameworks: 2020-01-15 16:18:03.334
xcodebuild[65297:3603150] unable to create build operation: Error
Domain=com.apple.xcbuild Code=1 "unknown error while handling message:
missingTarget(guid:
"17c6022e38d2bcbe4a9c6b3db6d054268267ba7180bd39bd8aedf115175222a2")"
UserInfo={NSLocalizedDescription=unknown error while handling message:
missingTarget(guid:
"17c6022e38d2bcbe4a9c6b3db6d054268267ba7180bd39bd8aedf115175222a2")}
** BUILD FAILED **

**Comment by mynameissujie** [comment]
flutter doctor:
[✓] Flutter (Channel master, v1.13.10-pre.65, on Mac OS X 10.14.4 18E226, locale
    zh-Hans-CN)
[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from:
      https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK
      components.
      (or visit https://flutter.dev/setup/#android-setup for detailed
      instructions).
      If the Android SDK has been installed to a custom location, set
      ANDROID_HOME to that location.
      You may also want to add it to your PATH environment variable.

[✓] Xcode - develop for iOS and macOS (Xcode 11.0)
[!] Android Studio (not installed)
[✓] VS Code (version 1.41.1)
[✓] Connected device (1 available)

**Comment by IchordeDionysos** [comment]
We've set up a sample add2app module from scratch, and it worked.
So we've thought that the dependencies might cause this bug.

So we gradually added more and more packages, and when we put in the line `cloud_firestore: ^0.13.0+1`, we started to see this issue to occur.

Interestingly, in the last time, we have many build problems with Firestore.
We were unable to build the app with [Option A](https://flutter.dev/docs/development/add-to-app/ios/project-setup#option-a---embed-with-cocoapods-and-the-flutter-sdk), we had hope that building Flutter in advance would magically solve the problems.

This may be related: FirebaseExtended/flutterfire#1853

**Comment by IchordeDionysos** [comment]
@jmagman Can you reproduce this? :) 

We are using add-to-app for quite some time now also with Firestore as a dependency and it mostly worked for us. Yes, sometimes there were problems including ones with Firestore, but until now we were always able to fix these by cleaning cache, deintegrating Cocoapods, etc. 🤔 

With this bug, we have tried everything we could but had no success so far even going back to old versions has not worked for us. We believe this might be a Firestore bug, but we aren't sure. 😄 

Initially, we had this bug, FirebaseExtended/flutterfire#1853, but then we found that we can directly build the add-to-app module we hoped that this would work, but we ended up with this issue here. 🙈 

It prevents us from releasing any app updates...

@jmagman Should we bring this to the attention of the Firebase developers and ask them to investigate the bug or is this something that has to be fixed on the Flutter side?

**Comment by jmagman** [comment]
I'm not sure what errors you're seeing without verbose logs (`flutter build ios-framework --output=build_ios_test -v`) but in my test Xcode is spewing errors about headers being copied multiple times:

```
1) Target 'BoringSSL-GRPC' (project 'Pods') has copy command from 'Pods/BoringSSL-GRPC/crypto/bio/internal.h' to 'Debug-iphoneos/BoringSSL-GRPC/openssl_grpc.framework/Headers/internal.h'
2) Target 'BoringSSL-GRPC' (project 'Pods') has copy command from 'Pods/BoringSSL-GRPC/crypto/bytestring/internal.h' to 'Debug-iphoneos/BoringSSL-GRPC/openssl_grpc.framework/Headers/internal.h'
3) Target 'BoringSSL-GRPC' (project 'Pods') has copy command from 'Pods/BoringSSL-GRPC/crypto/cipher_extra/internal.h' to 'Debug-iphoneos/BoringSSL-GRPC/openssl_grpc.framework/Headers/internal.h'
```
etc

This may be because the generated add-to-app Podfile is making all headers public:
```ruby
post_install do |installer|
  installer.pods_project.targets.each do |target|
    # Aggregate targets do not have a headers build phase.
    if target.respond_to?('headers_build_phase')
      target.headers_build_phase.files.each do |file|
        # Make headers public so they can be imported by the host application.
        file.settings = { 'ATTRIBUTES' => ['Public'] }
      end
    end
  end
end
```
And BoringSSL is trying on its own to copy headers around.
<img width="364" alt="Screen Shot 2020-02-04 at 1 26 06 PM" src="https://user-images.githubusercontent.com/682784/73788471-ee0d6600-4751-11ea-8e2e-5ae9d2a39248.png">
See their comment at:
https://github.com/grpc/grpc/blob/master/src/objective-c/BoringSSL-GRPC.podspec#L116-L122

**Comment by jmagman** [comment]
Going to try swapping out the Public headers Podfile hack with the new `BUILD_LIBRARY_FOR_DISTRIBUTION` flag.

**Comment by jmagman** [comment]
@IchordeDionysos I don't think you are hitting the same issue as @mynameissujie, but I am able to reproduce the issue and I think I have a fix.  I'm going to file a new issue.

@mynameissujie  Can we get reproduction steps?  I'm not sure what's going on with your plugin to give that error message.

**Comment by jmagman** [comment]
@IchordeDionysos I moved https://github.com/flutter/flutter/issues/50532.

**Comment by Lishaoshuai** [comment]
i have a same error, while i use shared_preferences plugin，i can't build ios-framework success.

**Comment by jmagman** [comment]
@Lishaoshuai Do you still see the issue on the [master channel](https://github.com/flutter/flutter/wiki/Flutter-build-release-channels#how-to-change-channels) after running `flutter clean`?

**Comment by jmagman** [comment]
This is a duplicate of https://github.com/flutter/flutter/issues/46993 which was fixed in 1.13.3.  The workaround is to pass in an absolute path in `--output`

```
flutter build ios-framework --output=/absolute/path/to/build_ios_test
```

Closing.
